dfsnew/registry/index.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2015 The Serviced Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package registry
+
+import (
+	"github.com/control-center/serviced/commons"
+	"github.com/control-center/serviced/commons/docker"
+	"github.com/control-center/serviced/datastore"
+	"github.com/control-center/serviced/domain/registry"
+)
+
+// RegistryIndex is the index for the docker registry on the server
+type RegistryIndex interface {
+	FindImage(image string) (*registry.Image, error)
+	PushImage(image, uuid string) error
+	RemoveImage(image string) error
+	SearchLibraryByTag(library string, tag string) ([]registry.Image, error)
+}
+
+var _ = RegistryIndex(&RegistryIndexClient{})
+
+// facade has the index for this docker registry
+type facade interface {
```

Comment:
Is the plan to move this interface into the facade package by the time the dust settles on this feature?